python/extensions/pip.bzl

```diff
@@ -316,9 +437,55 @@ Each `pip.parse()` call configures a particular Python version. Multiple calls
 can be made to configure different Python versions, and will be grouped by
 the `hub_name` argument. This allows the same logical name, e.g. `@pip//numpy`
 to automatically resolve to different, Python version-specific, libraries.
+
+pip.whl_mods:
+This tag class is used to create different wheel modification JSON files.  These files
+contain directives that are used by the wheel_installer.py during the creation of 
+wheels.
 """,
     implementation = _pip_impl,
     tag_classes = {
-        "parse": tag_class(attrs = _pip_parse_ext_attrs()),
+        "parse": tag_class(
+            attrs = _pip_parse_ext_attrs(),
+            doc = """\
+This tag class is used to create a pip hub and all of the spokes that are part of that hub.
+This tag class reuses the pip attributes that are found in 
+@rules_python//python/pip_install:pip_repository.bzl
+""",
+        ),
+        "whl_mods": tag_class(
+            attrs = _mod_tag_attrs(),
+            doc = """\
+This tag class is used to create JSON file that are used when calling wheel_builder.py.  These
+JSON files contain instructions on how to modify a wheel's project.  Each of the attributes
+create different modifications based on the type of attribute. Previously to bzlmod these
+JSON files where referred to as annotations, and were renamed to whl_modifications in this
+extension.
+""",
+        ),
+    },
+)
+
+def _whl_mods_repo_impl(rctx):
+    rctx.file("BUILD.bazel", """\
```

Comment:
These are only passed around as repo-level targets, so the contents of the build file don't matter. Just write an empty file; it makes our public api smaller. If a use case for rule-level accessing of them comes up, we can always add exports_files().